tests/integration/test.replication_events.js

```diff
@@ -170,5 +170,71 @@ adapters.forEach(function (adapters) {
       }).catch(done);
     });
 
+
+    // this test sets up a 2 way replication which initially transfers
+    // documents from a remote to a local database. Once the initial
+    // sync is complete, it inserts documents locally - the changes
+    // should propagate to the remote database and then back to the
+    // local database via the live replications. However, we do not
+    // expect any "change" events to fire for these documents since
+    // they already exist locally.
+    it('#4627 Test no duplicate changes in live replication', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        var remote = new PouchDB(dbs.remote);
+        var docId = -1;
+        var docsToGenerate = 10;
+        var lastChange = -1;
+
+        function generateDocs(n) {
+          return Array.apply(null, new Array(n)).map(function (e, i) {
+            docId += 1;
+            return {
+              _id: docId.toString(),
+              foo: Math.random().toString()
+            };
+          });
+        }
+
+        remote.bulkDocs(generateDocs(docsToGenerate)).then(function () {
+          db.replicate.to(remote, {
+            live: true,
+            retry: true,
+            since: 0
+          });
```

Comment:
So this is very tricky, I wrote a post about it @ http://pouchdb.com/2014/12/11/the-pains-of-being-async-at-heart.html (the last section), we can only finish this test once we are absolutely certain that both this and the previous replications are complete, its pretty much impossible to cleanup if not and it will lead to hard to detect intermittent failures. Once you are happy the test is complete then you need to `.cancel()` both the replications then call done when they have both fired their `complete` event handler, https://github.com/pouchdb/pouchdb/blob/master/tests/integration/test.replication.js#L534 may give you some help with that